### PR TITLE
CI: Add matrix to run tests in parallel

### PR DESCRIPTION
**Pull request by anhductn2001**
## Description

----

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

---;

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: .github/workflows/e2e_test.yml [hunk 1]
```diff
@@ -12,7 +12,10 @@ permissions:
   packages: write
 
 jobs:
-  test-ibc-transfer:
+  e2e-test:
+    strategy:
+      matrix:
+        tests: [e2e-test-ibc, e2e-test-ibc-timeout]
```
Comment (pgoos) [inline .github/workflows/e2e_test.yml:18]: When our test base grows, do we need to specify every test here?

**Review by pgoos — COMMENTED** [review]
Are we sure that it's working as intended? By looking at CI logs, I see that the 2nd test has ben triggered just after the first one finished. See https://github.com/dymensionxyz/dymension/actions/runs/7970087584/job/21757011998?pr=582

File: .github/workflows/e2e_test.yml (hunk 1 above)
Comment (anhductn2001) [reply .github/workflows/e2e_test.yml:18]: I think so, we will need to add our test case names here as we develop them. I might expand it to automatically collect these commands

**Comment by anhductn2001** [comment]
> Are we sure that it's working as intended? By looking at CI logs, I see that the 2nd test has ben triggered just after the first one finished. See https://github.com/dymensionxyz/dymension/actions/runs/7970087584/job/21757011998?pr=582

I'm sure they run in parallel

**Comment by anhductn2001** [comment]
I will close this PR because we will create another PR to move the e2e tests to a different repository.

**Pull request closed without merging**